[CI] EsqlSpecIT class failing

**Build Scans:**
- [elasticsearch-periodic-platform-support #6754 / ubuntu-2004-aarch64_checkpart3_platform-support-arm](https://gradle-enterprise.elastic.co/s/uvdn5b2hblrkg)
- [elasticsearch-periodic-platform-support #6733 / ubuntu-2404-aarch64_checkpart3_platform-support-arm](https://gradle-enterprise.elastic.co/s/vcszrn4rusae4)

**Reproduction Line:**
```
undefined
```

**Applicable branches:**
8.18

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.esql.qa.multi_node.EsqlSpecIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(org.elasticsearch.xpack.esql.qa.multi_node.EsqlSpecIT),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
undefined
```

**Issue Reasons:**
- [8.18] 2 failures in class org.elasticsearch.xpack.esql.qa.multi_node.EsqlSpecIT (1.7% fail rate in 119 executions)
- [8.18] 2 failures in pipeline elasticsearch-periodic-platform-support (50.0% fail rate in 4 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

I believe this is the same as https://github.com/elastic/elasticsearch/issues/120948.

The latter should've been fixed by https://github.com/elastic/elasticsearch/pull/121107. @dnhatn , do you have an idea what could cause the `seqNo [0] was processed twice in generation [2]` errors to still show up?

The first build scan has:
```
[2025-01-31T16:17:02,787][ERROR][o.e.b.ElasticsearchUncaughtExceptionHandler] [test-cluster-0] fatal error in thread [elasticsearch[test-cluster-0][generic][T#3]], exiting java.lang.AssertionError: java.lang.AssertionError: seqNo [0] was processed twice in generation [2], with different data. prvOp [Index{id='1', seqNo=0, primaryTerm=1, version=1, autoGeneratedIdTimestamp=-1}], newOp [Index{id='1', seqNo=0, primaryTerm=1, version=1, autoGeneratedIdTimestamp=-1}]	
```

The second build scan has this error:
```
[2025-01-28T14:55:50,156][ERROR][o.e.b.ElasticsearchUncaughtExceptionHandler] [test-cluster-1] fatal error in thread [elasticsearch[test-cluster-1][generic][T#21]], exiting java.lang.AssertionError: java.lang.AssertionError: seqNo [0] was processed twice in generation [2], with different data. prvOp [Index{id='1', seqNo=0, primaryTerm=1, version=1, autoGeneratedIdTimestamp=-1}], newOp [Index{id='1', seqNo=0, primaryTerm=1, version=1, autoGeneratedIdTimestamp=-1}]	
```

@dnhatn this got muted again. Marking as `blocker` again for the duration of the mute.

We got more info logged this time, though:
```
[2025-02-05T08:28:02,366][ERROR][o.e.b.ElasticsearchUncaughtExceptionHandler] [test-cluster-0] fatal error in thread [elasticsearch[test-cluster-0][generic][T#17]], exiting java.lang.AssertionError: java.lang.AssertionError: seqNo [0] was processed twice in generation [2], with different data. prvOp [Index{id='1', seqNo=0, primaryTerm=1, version=1, autoGeneratedIdTimestamp=-1} source: {st_logs=2024-12-23T12:15:00.000Z 1.2.3.4 [example@example.com](mailto:example@example.com) 4553, st_double=5.20128E11, st_geopoint=POINT(42.97109630194 14.7552534413725), _inference_fields={st_logs={inference={inference_id=test_sparse_inference, model_settings={task_type=sparse_embedding}, chunks={st_logs=[{start_offset=0, end_offset=57, embeddings={feature_0=9.994094E7, feature_1=9.994094E7, feature_2=9.994094E7, feature_3=9.994095E7, feature_4=9.994095E7}}]}}}, st_double={inference={inference_id=test_sparse_inference, model_settings={task_type=sparse_embedding}, chunks={st_double=[{start_offset=0, end_offset=10, embeddings={feature_0=1.02139264E9, feature_1=1.02139264E9, feature_2=1.02139264E9, feature_3=1.02139264E9, feature_4=1.02139264E9}}]}}}, st_geopoint={inference={inference_id=test_sparse_inference, model_settings={task_type=sparse_embedding}, chunks={st_geopoint=[{start_offset=0, end_offset=38, embeddings={feature_0=4.8845428E7, feature_1=4.8845428E7, feature_2=4.8845428E7, feature_3=4.884543E7, feature_4=4.884543E7}}]}}}, st_integer={inference={inference_id=test_sparse_inference, model_settings={task_type=sparse_embedding}, chunks={st_integer=[{start_offset=0, end_offset=2, embeddings={feature_0=1602.0, feature_1=1603.0, feature_2=1604.0, feature_3=1605.0, feature_4=1606.0}}]}}}, st_geoshape={inference={inference_id=test_sparse_inference, model_settings={task_type=sparse_embedding}, chunks={st_geoshape=[{start_offset=0, end_offset=45, embeddings={feature_0=2.9025277E8, feature_1=2.9025277E8, feature_2=2.9025277E8, feature_3=2.9025277E8, feature_4=2.9025277E8}}]}}}, st_datetime={inference={inference_id=test_sparse_inference, model_settings={task_type=sparse_embedding}, chunks={st_datetime=[{start_offset=0, end_offset=24, embeddings={feature_0=1.7914711E9, feature_1=1.7914711E9, feature_2=1.7914711E9, feature_3=1.7914711E9, feature_4=1.7914711E9}}]}}}, st_unicode={inference={inference_id=test_sparse_inference, model_settings={task_type=sparse_embedding}, chunks={st_unicode=[{start_offset=0, end_offset=5, embeddings={feature_0=2.0297848E9, feature_1=2.0297848E9, feature_2=2.0297848E9, feature_3=2.0297848E9, feature_4=2.0297848E9}}]}}}, st_unsigned_long={inference={inference_id=test_sparse_inference, model_settings={task_type=sparse_embedding}, chunks={st_unsigned_long=[{start_offset=0, end_offset=10, embeddings={feature_0=3.597711E8, feature_1=3.597711E8, feature_2=3.597711E8, feature_3=3.597711E8, feature_4=3.597711E8}}]}}}, st_ip={inference={inference_id=test_sparse_inference, model_settings={task_type=sparse_embedding}, chunks={st_ip=[{start_offset=0, end_offset=7, embeddings={feature_0=1.9016198E9, feature_1=1.9016198E9, feature_2=1.9016198E9, feature_3=1.9016198E9, feature_4=1.9016198E9}}]}}}, st_bool={inference={inference_id=test_sparse_inference, model_settings={task_type=sparse_embedding}, chunks={st_bool=[{start_offset=0, end_offset=5, embeddings={feature_0=9.719632E7, feature_1=9.719633E7, feature_2=9.719633E7, feature_3=9.719633E7, feature_4=9.719633E7}}]}}}, st_base64={inference={inference_id=test_sparse_inference, model_settings={task_type=sparse_embedding}, chunks={st_base64=[{start_offset=0, end_offset=12, embeddings={feature_0=1.4194486E9, feature_1=1.4194486E9, feature_2=1.4194486E9, feature_3=1.4194486E9, feature_4=1.4194486E9}}]}}}, st_multi_value={inference={inference_id=test_sparse_inference, model_settings={task_type=sparse_embedding}, chunks={st_multi_value=[{start_offset=0, end_offset=12, embeddings={feature_0=1.457005E8, feature_1=1.457005E8, feature_2=1.4570051E8, feature_3=1.4570051E8, feature_4=1.4570051E8}}, {start_offset=13, end_offset=35, embeddings={feature_0=8.689444E8, feature_1=8.689444E8, feature_2=8.689444E8, feature_3=8.689444E8, feature_4=8.689444E8}}, {start_offset=36, end_offset=56, embeddings={feature_0=1.5083834E9, feature_1=1.5083834E9, feature_2=1.5083834E9, feature_3=1.5083834E9, feature_4=1.5083834E9}}]}}}, st_long={inference={inference_id=test_sparse_inference, model_settings={task_type=sparse_embedding}, chunks={st_long=[{start_offset=0, end_offset=10, embeddings={feature_0=3.597711E8, feature_1=3.597711E8, feature_2=3.597711E8, feature_3=3.597711E8, feature_4=3.597711E8}}]}}}, semantic_text_field={inference={inference_id=test_sparse_inference, model_settings={task_type=sparse_embedding}, chunks={semantic_text_field=[{start_offset=0, end_offset=21, embeddings={feature_0=2.9864544E8, feature_1=2.9864544E8, feature_2=2.9864544E8, feature_3=2.9864544E8, feature_4=2.9864544E8}}]}}}, st_cartesian_point={inference={inference_id=test_sparse_inference, model_settings={task_type=sparse_embedding}, chunks={st…	
	at org.elasticsearch.server@9.1.0-SNAPSHOT/org.elasticsearch.index.translog.TranslogWriter.assertNoSeqNumberConflict(TranslogWriter.java:307)	
	at org.elasticsearch.server@9.1.0-SNAPSHOT/org.elasticsearch.index.translog.TranslogWriter.add(TranslogWriter.java:265)	
	at org.elasticsearch.server@9.1.0-SNAPSHOT/org.elasticsearch.index.translog.Translog.add(Translog.java:633)
```

--- a/x-pack/plugin/esql/qa/server/mixed-cluster/src/javaRestTest/java/org/elasticsearch/xpack/esql/qa/mixed/MixedClusterEsqlSpecIT.java
+++ b/x-pack/plugin/esql/qa/server/mixed-cluster/src/javaRestTest/java/org/elasticsearch/xpack/esql/qa/mixed/MixedClusterEsqlSpecIT.java
@@ -71,11 +71,6 @@ public class MixedClusterEsqlSpecIT extends EsqlSpecTestCase {
     }
 
     @Override
-    protected boolean shouldSkipTestsWithSemanticTextFields() {
-        return true;
-    }
-
-    @Override
     protected boolean enableRoundingDoubleValuesOnAsserting() {
         return true;
     }

--- a/x-pack/plugin/esql/qa/server/multi-clusters/src/javaRestTest/java/org/elasticsearch/xpack/esql/ccq/MultiClusterSpecIT.java
+++ b/x-pack/plugin/esql/qa/server/multi-clusters/src/javaRestTest/java/org/elasticsearch/xpack/esql/ccq/MultiClusterSpecIT.java
@@ -132,11 +132,6 @@ public class MultiClusterSpecIT extends EsqlSpecTestCase {
         assumeFalse("UNMAPPED FIELDS not yet supported in CCS", testCase.requiredCapabilities.contains(UNMAPPED_FIELDS.capabilityName()));
     }
 
-    @Override
-    protected boolean shouldSkipTestsWithSemanticTextFields() {
-        return true;
-    }
-
     private TestFeatureService remoteFeaturesService() throws IOException {
         if (remoteFeaturesService == null) {
             var remoteNodeVersions = readVersionsFromNodesInfo(remoteClusterClient());

--- a/x-pack/plugin/esql/qa/server/single-node/src/javaRestTest/java/org/elasticsearch/xpack/esql/qa/single_node/EsqlSpecIT.java
+++ b/x-pack/plugin/esql/qa/server/single-node/src/javaRestTest/java/org/elasticsearch/xpack/esql/qa/single_node/EsqlSpecIT.java
@@ -44,11 +44,6 @@ public class EsqlSpecIT extends EsqlSpecTestCase {
     }
 
     @Override
-    protected boolean shouldSkipTestsWithSemanticTextFields() {
-        return cluster.getNumNodes() > 1;
-    }
-
-    @Override
     protected boolean supportsSourceFieldMapping() {
         return cluster.getNumNodes() == 1;
     }

--- a/x-pack/plugin/esql/qa/server/src/main/java/org/elasticsearch/xpack/esql/qa/rest/EsqlSpecTestCase.java
+++ b/x-pack/plugin/esql/qa/server/src/main/java/org/elasticsearch/xpack/esql/qa/rest/EsqlSpecTestCase.java
@@ -70,7 +70,6 @@ import static org.elasticsearch.xpack.esql.CsvTestsDataLoader.createInferenceEnd
 import static org.elasticsearch.xpack.esql.CsvTestsDataLoader.deleteInferenceEndpoint;
 import static org.elasticsearch.xpack.esql.CsvTestsDataLoader.loadDataSetIntoEs;
 import static org.elasticsearch.xpack.esql.EsqlTestUtils.classpathResources;
-import static org.elasticsearch.xpack.esql.action.EsqlCapabilities.Cap.SEMANTIC_TEXT_TYPE;
 import static org.elasticsearch.xpack.esql.action.EsqlCapabilities.Cap.SOURCE_FIELD_MAPPING;
 
 // This test can run very long in serverless configurations
@@ -130,9 +129,6 @@ public abstract class EsqlSpecTestCase extends ESRestTestCase {
 
     @Before
     public void setup() throws IOException {
-        if (shouldSkipTestsWithSemanticTextFields()) {
-            assumeFalse("semantic_text tests are muted", testCase.requiredCapabilities.contains(SEMANTIC_TEXT_TYPE.capabilityName()));
-        }
         if (supportsInferenceTestService() && clusterHasInferenceEndpoint(client()) == false) {
             createInferenceEndpoint(client());
         }
@@ -142,11 +138,6 @@ public abstract class EsqlSpecTestCase extends ESRestTestCase {
         if (indexExists(availableDatasetsForEs(client(), supportsLookup, supportsSourceMapping).iterator().next().indexName()) == false) {
             loadDataSetIntoEs(client(), supportsLookup, supportsSourceMapping);
         }
-    }
-
-    // https://github.com/elastic/elasticsearch/issues/121411
-    protected boolean shouldSkipTestsWithSemanticTextFields() {
-        return false;
     }
 
     @AfterClass
@@ -184,9 +175,6 @@ public abstract class EsqlSpecTestCase extends ESRestTestCase {
         }
         checkCapabilities(adminClient(), testFeatureService, testName, testCase);
         assumeTrue("Test " + testName + " is not enabled", isEnabled(testName, instructions, Version.CURRENT));
-        if (shouldSkipTestsWithSemanticTextFields()) {
-            assumeFalse("semantic_text tests are muted", testCase.requiredCapabilities.contains(SEMANTIC_TEXT_TYPE.capabilityName()));
-        }
         if (supportsSourceFieldMapping() == false) {
             assumeFalse("source mapping tests are muted", testCase.requiredCapabilities.contains(SOURCE_FIELD_MAPPING.capabilityName()));
         }

